src/plugins/MemoryKeyStore.ts

```diff
@@ -56,6 +56,33 @@ export class MemoryKeyStore implements KeyStore {
     return Promise.resolve(metadata);
   }
 
+  public updateKeys(keys: EncryptedKey[]) {
+    return Promise.all(
+      keys.map((encryptedKey: EncryptedKey) => {
+        const publicKey = encryptedKey.publicKey;
+        if (!this.keyStore[publicKey]) {
+          return Promise.reject(publicKey);
+        }
+
+        const creationTime = this.keyStore[publicKey].creationTime;
+        const modifiedTime = getNow();
+        this.keyStore[publicKey] = {
+          creationTime,
+          modifiedTime,
+          encryptedKey,
+        };
+
+        return Promise.resolve(
+          getMetadataFromMemoryItem({
+            encryptedKey,
+            creationTime,
+            modifiedTime,
+          }),
+        );
+      }),
+    );
+  }
+
```

Comment:
I could be wrong but I think this is not atomic, since if one of the keys is not in the store, that one will reject but the others will update, and the function will reject even though some items got updated.

I think we should either require atomicity or not, and if we don't, we shouldn't mention it.

Reply:
I think that's right. It would be good to update this so that if one fails, they all fail.